### PR TITLE
MAT-9583 allow editing of components in composite measure

### PR DESCRIPTION
**Pull request by sb-prateekkeerthi**
## MADiE PR

Jira Ticket: [MAT-9583](https://jira.cms.gov/browse/MAT-9583)
(Optional) Related Tickets:

### Summary
allow editing of components in composite measure

### All Submissions
* [ ] This PR has the JIRA linked.
* [ ] Required tests are included.
* [ ] No extemporaneous files are included (i.e Complied files or testing results).
* [ ] This PR is merging into the **correct branch**.
* [ ] All Documentation needed for this PR is Complete (or noted in a TODO or other Ticket).
* [ ] Any breaking changes or failing automations are noted by placing a comment on this PR.

### DevSecOps
If there is a question if this PR has a security or infrastructure impact, please contact the Security or DevOps engineer assigned to this project to discuss it further.

* [ ] This PR has NO significant security impact (i.e Changing auth methods, Adding a new user type, Adding a required but vulnerable package).
* [ ] All CDN/Web dependencies are hosted internally (i.e MADiE-Root Repo).

### Reviewers
By Approving this PR you are attesting to the following:

*  Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose.
*  The tests appropriately test the new code, including edge cases.
*  If you have any concerns they are brought up either to the developer assigned, security engineer, or leads.

### CODE REVIEW

File: src/components/editMeasure/populationCriteria/groups/Composite/addComponents/AddComponentsDialog.tsx [hunk 1]
```diff
@@ -297,6 +338,7 @@ export default function AddComponentsDialog({
     selectedIdForExpansion,
     isRowExpanded,
     expandedSectionData,
+    components,
```
Comment (etan-sb) [inline src/components/editMeasure/populationCriteria/groups/Composite/addComponents/AddComponentsDialog.tsx]: I think the only two dependencies need to be `selectedIdForExpansion` and `isRowExpanded` since the rest are never referenced in the useMemo

File: src/components/editMeasure/populationCriteria/groups/Composite/compositeScoring/AddedComponentsTable.tsx [hunk 2]
```diff
@@ -30,12 +30,16 @@ const TH = tw.th`p-3 text-left text-sm font-bold capitalize`;
 
 export default function AddedComponentsTable({
   components,
+  onComponentsUpdate,
 }: {
   components: Component[];
+  onComponentsUpdate: (updatedComponents: Component[]) => void;
+  measureId: string;
```
Comment (etan-sb) [inline src/components/editMeasure/populationCriteria/groups/Composite/compositeScoring/AddedComponentsTable.tsx]: Unused prop? Maybe for debugging or future story?

File: src/components/editMeasure/populationCriteria/groups/Composite/compositeScoring/AddedComponentsTable.tsx [hunk 3]
```diff
@@ -60,6 +64,21 @@ export default function AddedComponentsTable({
     fetchMeasuresForComponents();
   }, [fetchMeasuresForComponents]);
 
+  const handleDeleteComponent = async (measureId: string) => {
```
Comment (etan-sb) [inline src/components/editMeasure/populationCriteria/groups/Composite/compositeScoring/AddedComponentsTable.tsx]: Does this function need to be `async`? Don't see any `await` statements.

File: src/components/editMeasure/populationCriteria/groups/Composite/compositeScoring/AddedComponentsTable.tsx (hunk 2 above)
Comment (sb-prateekkeerthi) [reply src/components/editMeasure/populationCriteria/groups/Composite/compositeScoring/AddedComponentsTable.tsx]: looks like I forgot to remove it, thanks for letting me know

File: src/components/editMeasure/populationCriteria/groups/Composite/compositeScoring/AddedComponentsTable.tsx (hunk 3 above)
Comment (sb-prateekkeerthi) [reply src/components/editMeasure/populationCriteria/groups/Composite/compositeScoring/AddedComponentsTable.tsx]: looks like I forgot to remove it, thanks for letting me know

File: src/components/editMeasure/populationCriteria/groups/Composite/addComponents/AddComponentsDialog.tsx (hunk 1 above)
Comment (sb-prateekkeerthi) [reply src/components/editMeasure/populationCriteria/groups/Composite/addComponents/AddComponentsDialog.tsx]: thank you, yeah I have removed it and pushed the changes

**Review by etan-sb — APPROVED** [review]
